src/devices/LedDisplay/Large4Digit7SegmentDisplay.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Runtime.InteropServices;
+using Iot.Device.LedDisplay.Enums;
+
+namespace Iot.Device.LedDisplay
+{
+    /// <summary>
+    /// Adafruit 1.2" 4-Digit 7-Segment Display w/I2C Backpack
+    /// </summary>
+    /// <remarks>
+    /// Comes in yellow, green and red colors:
+    /// https://www.adafruit.com/product/1268
+    /// https://www.adafruit.com/product/1269
+    /// https://www.adafruit.com/product/1270
+    /// Sources:
+    /// https://github.com/adafruit/Adafruit_LED_Backpack/blob/master/Adafruit_LEDBackpack.cpp
+    /// https://github.com/sobek1985/Adafruit_LEDBackpack/blob/master/Adafruit_LEDBackpack/AlphaNumericFourCharacters.cs
+    /// </remarks>
+    public class Large4Digit7SegmentDisplay : Ht16k33, ISevenSegmentDisplay
+    {
+        #region Private members
+        #region Constants
+        /// <summary>
+        /// Number of digits supported by display
+        /// </summary>
+        private const int MaxNumberOfDigits = 4;
+
+        /// <summary>
+        /// This display does not support dot bits for each digit
+        /// </summary>
+        private const byte SegmentMask = 0b0111_1111;
+        #endregion
+
+        #region Enums
+        /// <summary>
+        /// Digit address within display buffer
+        /// </summary>
+        private enum Address
+        {
+            /// <summary>
+            /// First digit
+            /// </summary>
+            Digit1 = 1,
+            /// <summary>
+            /// Second digit
+            /// </summary>
+            Digit2 = 3,
+            /// <summary>
+            /// Dot setting bits
+            /// </summary>
+            Dots = 5,
+            /// <summary>
+            /// Third digit
+            /// </summary>
+            Digit3 = 7,
+            /// <summary>
+            /// Fourth digit
+            /// </summary>
+            Digit4 = 9
+        }
+        #endregion
+
+        /// <summary>
+        /// List of digit addresses for sequential writing
+        /// </summary>
+        private static readonly Address[] s_digitAddressList = new[] { Address.Digit1, Address.Digit2, Address.Digit3, Address.Digit4 };
+
+        /// <summary>
+        /// Empty display buffer
+        /// </summary>
+        private static readonly byte[] s_clearBuffer = new byte[]
+        {
+            0x00, 0b0000_0000,
+            0x01, 0b0000_0000,
+            0x02, 0b0000_0000,
+            0x03, 0b0000_0000,
+            0x04, 0b0000_0000
+        };
+
+        /// <summary>
+        /// Display buffer
+        /// </summary>
+        private readonly byte[] _displayBuffer = new byte[]
```

Comment:
should this one be static too? also since this is the same as `s_clearBuffer` can you assign it such as well? (you can write a comment that this is incidental)

Reply:
If the display buffer's static then you can only control a single device within one process.